cloud/pkg/synccontroller/synccontroller.go

```diff
@@ -222,6 +240,59 @@ func (sctl *SyncController) manageObjectSync(syncs []*v1alpha1.ObjectSync) {
 		}
 	}
 }
+func (sctl *SyncController) DeleteOutdatedObjectSync() {
+	for {
+		select {
+		case <-beehiveContext.Done():
+			klog.Warning("Stop synccontroller DeleteOutdatedObjectSync loop")
+			return
+		case e := <-sctl.nodeManager.Events():
+			node, ok := e.Object.(*v1.Node)
+			if !ok {
+				klog.Warningf("Object type: %T unsupported", node)
+				continue
+			}
+			switch e.Type {
+			case watch.Deleted:
+				syncs, err := sctl.objectSyncLister.List(labels.Everything())
+				if err != nil {
+					klog.Errorf("Filed to list all the ObjectSyncs: %v", err)
+				}
+				for _, sync := range syncs {
```

Comment:
What's the relationship of the sync and node here? Maybe the code needs some refactor.

Reply:
@fisherxu syncs refers to syncObject ,the relationship is the node have syncs.  when node unregisters, it's syncs should be deleted